api/importer/package_processor.go

```diff
@@ -101,6 +108,57 @@ func (ipp *ImportPackageProcessor) Process(project string, ip ImportPackage) err
 	return nil
 }
 
+func (ipp *ImportPackageProcessor) validateManifest(
+	manifest *model.ImportManifest, ip ImportPackage) error {
+	re := regexp.MustCompile("^[a-zA-Z0-9_]*$")
+
+	for _, task := range manifest.Tasks {
+		// Check if ID is not empty and formatted properly
+		if task.ID == "" {
+			return fmt.Errorf("task id cannot be empty")
+		} else if !re.MatchString(task.ID) {
+			return fmt.Errorf("task id %s can only consist of alphnumeric characters and underscores", task.ID)
+		}
+
+		// Check if task definition is correct
+		switch task.Type {
+		case apiTaskType:
+			if task.APITask != nil {
+				// Check if the action type is supported
+				if !slices.Contains(model.AllActions, task.APITask.Action) {
```

Comment:
We could move the constants from `execute` to the `executor` and access them over the injected executor, or do you have another idea for implementing this?

Reply:
I would expose a method from the executor interface like

```
ActionSupported(actionName string) bool
```
so that this can be checked before manifest execution but also without having to know all the action at processor level.
Furthermore the fact that the constant exists does not guarantee that the implementation is available ;) so the implementor of the interface can also check that the action is actually supported